Sources/Core/GTMSessionFetcherLogging.m

```diff
@@ -75,141 +77,165 @@ @implementation GTMSessionFetcher (GTMSessionFetcherLogging)
 static NSString *gLoggingProcessName = nil;
 
 + (void)setLoggingDirectory:(NSString *)path {
-  gLoggingDirectoryPath = [path copy];
+  @synchronized([GTMSessionFetcher class]) {
+    gLoggingDirectoryPath = [path copy];
+  }
 }
 
 + (NSString *)loggingDirectory {
-  if (!gLoggingDirectoryPath) {
-    NSArray *paths = nil;
+  @synchronized([GTMSessionFetcher class]) {
+    if (!gLoggingDirectoryPath) {
+      NSArray *paths = nil;
 #if TARGET_IPHONE_SIMULATOR
-    // default to a directory called GTMHTTPDebugLogs into a sandbox-safe
-    // directory that a developer can find easily, the application home
-    paths = @[ NSHomeDirectory() ];
+      // default to a directory called GTMHTTPDebugLogs into a sandbox-safe
+      // directory that a developer can find easily, the application home
+      paths = @[ NSHomeDirectory() ];
 #elif TARGET_OS_IPHONE
-    // Neither ~/Desktop nor ~/Home is writable on an actual iOS, watchOS, or tvOS device.
-    // Put it in ~/Documents.
-    paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
+      // Neither ~/Desktop nor ~/Home is writable on an actual iOS, watchOS, or tvOS device.
+      // Put it in ~/Documents.
+      paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
 #else
-    // default to a directory called GTMHTTPDebugLogs in the desktop folder
-    paths = NSSearchPathForDirectoriesInDomains(NSDesktopDirectory, NSUserDomainMask, YES);
+      // default to a directory called GTMHTTPDebugLogs in the desktop folder
+      paths = NSSearchPathForDirectoriesInDomains(NSDesktopDirectory, NSUserDomainMask, YES);
 #endif
 
-    NSString *desktopPath = paths.firstObject;
-    if (desktopPath) {
-      NSString *const kGTMLogFolderName = @"GTMHTTPDebugLogs";
-      NSString *logsFolderPath = [desktopPath stringByAppendingPathComponent:kGTMLogFolderName];
+      NSString *desktopPath = paths.firstObject;
+      if (desktopPath) {
+        NSString *const kGTMLogFolderName = @"GTMHTTPDebugLogs";
+        NSString *logsFolderPath = [desktopPath stringByAppendingPathComponent:kGTMLogFolderName];
+
+        NSFileManager *fileMgr = [NSFileManager defaultManager];
+        BOOL isDir;
+        BOOL doesFolderExist = [fileMgr fileExistsAtPath:logsFolderPath isDirectory:&isDir];
```

Comment:
Instead of checking and then creating, why don't you just create it every time? Get rid of a minor race condition.

Reply:
This is the existing flow, so I'd sorta like to not make other changes at the same time we fold in locks.  We can always revisit it in the future, but minimizing scope of changes per commit seems worth while.  Thoughts?